### DISCUSSION TITLE
Self hostes gateway only use container solution?

### DISCUSSION THREAD

**Discussion by edwardkuo** [Azure/api-management-self-hosted-gateway#143] [Q&A]
I find the architecture of the API management gateway is very similar to the mulesoft API management. It also uses the Linux containers to do its gateway solution. I ask them if there is any other solution if not the container solution. They also provide the install application solution to do its gateway. I am not sure API management gateway  is a similar solution

**Comment by tomkerkhove** [comment]
As of today, you can only run the self-hosted gateway by using containers.

Adding @ferantoMSFT who is our PM on this

**Reply by edwardkuo** [reply]
@tomkerkhove Yes, I know the self-hosted gateway only has the container solution.
But Is it impossible to use a self-hosted gateway without container or k8s?  I have asked mulesoft if they can provide a non-container solution for their API self-hosted gateway.

In theory, the self-hosted gateway can also be installed independently.

**Reply by tomkerkhove** [reply]
No, that was what I was mentioning - You have to use containers as of today.
